R/source.R

```diff
@@ -0,0 +1,31 @@
+
+
+#' Read and evaluate a Python script
+#'
+#' Evaluate a Python script and make created Python objects available within R.
+#'
+#' @inheritParams py_run_file
+#'
+#' @param envir The environment to assign Python objects into
+#'   (for example, `parent.frame()` or `globalenv()`). Specify `NULL` to
+#'   not assign Python objects.
+#'
+#' @export
+source_python <- function(file, envir = parent.frame(), convert = TRUE) {
+  
+  # source the python script
+  dict <- py_run_file(file, local = TRUE, convert = convert)
+  
+  # get the keys
+  names <- py_dict_get_keys_as_str(dict)
+  Encoding(names) <- "UTF-8"
+  
+  # assign the objects into the specified environment
+  if (!is.null(envir)) {
+    for (name in names)
+      assign(name, dict[[name]], envir = envir)
+  }
+  
+  # return nothing
+  invisible(NULL)
```

Comment:
Any reason why we return `NULL` rather than the dictionary produced by `py_run_file()`? It might be useful to enable:

```R
output <- source_python("file.py", NULL)
```

Reply:
The dictionary by itself might not be that useful (because they need to use Python dictionary methods to interact with it). We could copy it to an R list but that would create an extra copy operation that would often be unnecessary. My thinking was that we should emulate the R `source()` function and that as you said `py_run_file()` could be used for explicitly fetching the results of execution.

Reply:
Got it; sounds good!